aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -592,13 +592,20 @@ Tensor& diff_out(const Tensor& self, int64_t n, int64_t dim, const c10::optional
 
 // ALL REDUCE #################################################################
 
-static ScalarType get_dtype(Tensor& result, const Tensor& self, optional<ScalarType> dtype,
-                            bool promote_integers=false) {
+inline ScalarType get_dtype(Tensor& result, optional<ScalarType> dtype) {
```

Comment:
The new set of overloads here seems incredibly error prone here. `get_dtype(result, dtype)` treats result correctly, but `get_dtype(result, dtype, false)` treats result as `self`. Better to name them differently...